### ISSUE TITLE
MIDI Playback Issue (ADLMidi)

### ISSUE THREAD

**Issue by ItsT3K** [droidmjt/Droidsound#590]
Greetings,

I have been using Droidsound-E for quite some time and I've noticed on recent versions some MIDIs using AdlMidi seem to only play the Percussion channel and nothing else. These MIDIs play just fine on regular AdlMidi and Foobar2000 using AdlMidi but for some reason don't play correctly on Droidsound-E.

I'm using a Samsung Galaxy A71 5G UW (Qualcomm) with Android 10.

some of the midis in question come from bisqwits own website and are 
ct-frog.mid
kctgg-flaggame.mid
kor-06.mid

along with a few others.

Sorry in advance if it turns out to just be my phone but this is a new problem to me.

**Comment by droidmjt** [comment]
could you at least include the files instead of the filenames please :)

**Comment by droidmjt** [comment]
When this problem appeared or has it always been like this?

**Comment by droidmjt** [comment]
Also, what settings you have for ADLMIDI, which frequency, which bank, how many chips?

**Comment by ItsT3K** [comment]
Latest APK 64bit from download

Default ADLMidi settings Patch 1 (Bisqwit)

The midis in question are

https://bisqwit.iki.fi/music/misc/spc/kctgg-flaggame.mid

https://bisqwit.iki.fi/music/misc/spc/kor-06.mid

https://bisqwit.iki.fi/music/misc/spc/ct-frog.mid

(i could not figure out how to upload from mobile. sorry)

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/sktzyyp4kdiyxk6/droidsounde167_233_32bit.apk?dl=1
https://www.dropbox.com/s/mmidd2lwk47hj6z/droidsounde167_233_64bit.apk?dl=1
try if it works now

**Comment by ItsT3K** [comment]
They play the melodies now but they sound off from the actual output from adlmidi (via linux). The only one that sounds very similar to the pc output is ct-frog.mid

(Sorry if I sound like a jerk. I dont intend to come off as a jerk)

**Comment by droidmjt** [comment]
https://www.dropbox.com/s/q8kp6p7d5uk6wso/droidsounde167_234_32bit.apk?dl=1
https://www.dropbox.com/s/1tfdn6aaaqjrpt7/droidsounde167_234_64bit.apk?dl=1
try if it works now

**Comment by ItsT3K** [comment]
Yes! They sound exactly like the PC counterpart does. thank you!.

**Comment by Wohlstand** [comment]
@ItsT3K , One small note that AdlMidi for foobar2000 in some cases has two variants (one ancient, and some modern), the modern variant does use my fork of ADLMIDI, called as libADLMIDI https://github.com/Wohlstand/libADLMIDI
It's majorely different from ADLMIDI by Bisqwit because of a big set of improvments and polishes over it. One at Droidsound-E does use my library, however @droidmjt doen't care about keeping it's up to date.

Recently I made a new release of this library, see what was changed on it:
https://github.com/Wohlstand/libADLMIDI#changelog

Also, a small note about Bisqwit's MIDI files: they are messy because they are a result of SPC to MIDI conversion with no manual clean-up. Any MIDI files produced by a conversion from alien formats are oftenly require the manual clean-up and improvement to make them sound properly everywhere.
